File: src/main/java/rk/cinema/configuration/ReservationConfig.java
package rk.cinema.configuration;

import io.micronaut.context.annotation.ConfigurationProperties;
import jakarta.validation.constraints.Min;

@ConfigurationProperties("csrs.reservation")
public class ReservationConfig {

    @Min(1)
    private int timeoutMinutes;

    public int getTimeoutMinutes() {
        return timeoutMinutes;
    }

    public void setTimeoutMinutes(int timeoutMinutes) {
        this.timeoutMinutes = timeoutMinutes;
    }
}

File: src/main/kotlin/rk/powermilk/cinema/configuration/ReservationConfig.kt
package rk.powermilk.cinema.configuration

import io.micronaut.context.annotation.ConfigurationProperties
import jakarta.validation.constraints.Min

@ConfigurationProperties("csrs.reservation")
class ReservationConfig {
    @JvmField
    var timeoutMinutes: @Min(1) Int = 0
}
